src/functions-reference/higher-order_functions.Rmd

```diff
@@ -7,173 +7,148 @@ by name to the higher-order functions.
 
 An algebraic system is specified as an ordinary function in Stan
-within the function block. The algebraic system function must have
-this signature:
+within the function block. The function must return a `vector`
+and takes in, as its first argument, the unknowns $y$ we wish to solve for,
+also passed as a `vector`.
+This argument is followed by additional arguments as specified by the user;
+we  call such arguments _variadic arguments_ and denote them `...`.
+The signature of the algebraic system is then:
 
 ```stan
- vector algebra_system(vector y, vector theta,
-                              data array[] real x_r, array[] int x_i)
+ vector algebra_system (vector y, ...)
 ```
 
-The algebraic system function should return the value of the algebraic
-function which goes to 0, when we plug in the solution to the
-algebraic system.
-
-The argument of this function are:
-
-*   *`y`*, the unknowns we wish to solve for
-
-*   *`theta`*, parameter values used to evaluate the algebraic system
-
-*   *`x_r`*, data values used to evaluate the algebraic system
-
-*   *`x_i`*, integer data used to evaluate the algebraic system
-
-The algebraic system function separates parameter values, *`theta`*,
-from data values, *`x_r`*, for efficiency in propagating the derivatives
-through the algebraic system.
-
 ### Call to the algebraic solver
 
-<!-- vector; algebra_solver; (function algebra_system, vector y_guess, vector theta, data array[] real x_r, array[] int x_i); -->
-\index{{\tt \bfseries algebra\_solver }!{\tt (function algebra\_system, vector y\_guess, vector theta, data array[] real x\_r, array[] int x\_i): vector}|hyperpage}
-
-`vector` **`algebra_solver`**`(function algebra_system, vector y_guess, vector theta, data array[] real x_r, array[] int x_i)`<br>\newline
-Solves the algebraic system, given an initial guess, using the Powell
-hybrid algorithm.
-`r since("2.17")`
-
-<!-- vector; algebra_solver; (function algebra_system, vector y_guess, vector theta, data array[] real x_r, array[] int x_i, data real rel_tol, data real f_tol, int max_steps); -->
-\index{{\tt \bfseries algebra\_solver }!{\tt (function algebra\_system, vector y\_guess, vector theta, data  array[] real x\_r, array[] int x\_i, data real rel\_tol, data real f\_tol, int max\_steps): vector}|hyperpage}
+`vector` **`solve_newton`**`(function algebra_system, vector y_guess, ...)`<br>\newline
+Solves the algebraic system, given an initial guess, using Newton's method.
+`r since("2.31")`
 
-`vector` **`algebra_solver`**`(function algebra_system, vector y_guess, vector theta, data array[] real x_r, array[] int x_i, data real rel_tol, data real f_tol, int max_steps)`<br>\newline
-Solves the algebraic system, given an initial guess, using the Powell
-hybrid algorithm with additional control parameters for the solver.
-`r since("2.17")`
+`vector` **`solve_newton_tol`**`(function algebra_system, vector y_guess, data real scaling_step, data real f_tol, int max_steps)`<br>\newline
```

Comment:
Both `_tol` signatures are missing the `...`

Reply:
Good catch. Fixed!